kin-sdk/kin-sdk-lib/src/test/java/kin/sdk/TransactionTest.java

```diff
@@ -0,0 +1,100 @@
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 23, manifest = Config.NONE)
+public class TransactionTest {
+
+    private static final String ACCOUNT_ID_FROM = "GDJVKIY3UQKMTQZHXR36ZQUNFYO45XLM6IHWO6TYQ53M5KEXBNMJYWVR";
+    private static final String SECRET_SEED_FROM = "SB73L5FFTZMN6FHTOOWYEBVFTLUWQEWBLSCI4WLZADRJWENDBYL6QD6P";
+
+    private MockWebServer mockWebServer;
+    private Transaction transaction;
+
+    @Before
+    public void setup() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        mockServer();
+        Network.useTestNetwork();
+
+        createTransaction();
+
+    }
+
+    private void createTransaction() {
+        KeyPair source = KeyPair.fromSecretSeed(SECRET_SEED_FROM);
+        KeyPair destination = KeyPair.fromAccountId(ACCOUNT_ID_FROM);
+        long sequenceNumber = 2908908335136768L;
+        Account account = new Account(source, sequenceNumber);
+        kin.base.Transaction baseTransaction = new kin.base.Transaction.Builder(account)
+            .addOperation(new CreateAccountOperation.Builder(destination, "2000").build())
+            .addFee(100)
+            .build();
+        baseTransaction.sign(source);
+        TransactionId id = new TransactionIdImpl(Utils.byteArrayToHex(baseTransaction.hash()));
+        transaction = new Transaction(id, baseTransaction);
+    }
+
+    private void mockServer() throws IOException {
+        mockWebServer = new MockWebServer();
+        mockWebServer.start();
+    }
+
+    @Test
+    public void getTransactionEnvelope_success() throws Exception {
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_from.json"));
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_to.json"));
+
+        String transactionEnvelope = transaction.getTransactionEnvelope();
+        kin.base.Transaction transaction2 = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+
+        assertThat(transaction.getBaseTransaction().getSourceAccount().getAccountId(),
+            equalTo(transaction2.getSourceAccount().getAccountId()));
+        assertThat(transaction.getBaseTransaction().getSequenceNumber(), equalTo(transaction2.getSequenceNumber()));
+        assertThat(transaction.getBaseTransaction().getFee(), equalTo(transaction2.getFee()));
+        assertThat(transaction.getTransactionEnvelope(), equalTo(transaction2.toEnvelopeXdrBase64()));
+    }
+
+    @Test
+    public void decodeTransaction_success() throws Exception {
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_from.json"));
+        mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_to.json"));
+
+        String transactionEnvelope = transaction.getTransactionEnvelope();
+        Transaction transaction2 = Transaction.decodeTransaction(transactionEnvelope);
```

Comment:
same here but opposite, no need to build a transaction, build one externally and hardcode it here as an input for decode

Reply:
same comment as above